test/drivers/test_cqrrpt_gpu.cu

```diff
@@ -0,0 +1,257 @@
+
+    RandBLAS::DenseDist D1(m, n);
+    RandBLAS::DenseDist D2(n, n);
+    state = RandBLAS::fill_dense(D1, A.data(), state).second;
+    state = RandBLAS::fill_dense(D2, B.data(), state).second;
+
+    blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, n, n, 1.0, A.data(), m, B.data(), n, 1.0, C.data(), m);
+    char name1 [] = "C host";
+    RandBLAS::util::print_colmaj(m, n, C.data(), name1);
+
+    int deviceCount = 0;
+    cudaGetDeviceCount(&deviceCount);
+    printf("Device count: %d\n", deviceCount);
+
+    blas::Queue blas_queue(0);
+    double* A_device;
+    double* B_device;
+    double* C_device;
+    cudaMalloc(&A_device, m * n * sizeof(double));
+    cudaMalloc(&B_device, n * n * sizeof(double));
+    cudaMalloc(&C_device, m * n * sizeof(double));
+
+    cudaMemcpy(A_device, A.data(), m * n * sizeof(double), cudaMemcpyHostToDevice);
+    cudaMemcpy(B_device, B.data(), n * n * sizeof(double), cudaMemcpyHostToDevice);
+
+    blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, n, n, 1.0, A_device, m, B_device, n, 1.0, C_device, m, blas_queue);
+    //blas_queue.sync();
+
+    cudaMemcpy(C_received.data(), C_device,  m * n * sizeof(double), cudaMemcpyDeviceToHost);
+
+    cudaFree(A_device);
+    cudaFree(B_device);
+    cudaFree(C_device);
+
+    char name2 [] = "C received device";
+    RandBLAS::util::print_colmaj(m, n, C_received.data(), name2);
+}
+
+TEST_F(TestCQRRPT, something1) {
+    // Matrix dimensions
+    int m = 3; // Number of rows
+    int n = 2; // Number of columns
+
+    // Host matrices
+    std::vector<double> A = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f};
+    std::vector<double> B = {7.0f, 8.0f, 9.0f, 10.0f, 11.0f, 12.0f};
+
+    // Host workspace and info
+    std::vector<double> work;
+    int lwork_ormqr = 0;
+
+    int *devInfo = nullptr;
+
+    // CUDA device pointers
+    double *d_A = nullptr;
+    double *d_B = nullptr;
+    double *d_tau = nullptr;
+
+    // CUDA handles
+    cusolverDnHandle_t cusolverH = nullptr;
+
+    // Create cusolver handle
+    cusolverDnCreate(&cusolverH);
+
+    // Allocate CUDA memory for matrices and workspace
+    cudaMalloc((void**)&d_A, m * n * sizeof(double));
+    cudaMalloc((void**)&d_B, m * n * sizeof(double));
+    cudaMalloc((void**)&devInfo, sizeof(int));
+
+    // Copy matrices from host to device
+    cudaMemcpy(d_A, A.data(), m * n * sizeof(double), cudaMemcpyHostToDevice);
+    cudaMemcpy(d_B, B.data(), m * n * sizeof(double), cudaMemcpyHostToDevice);
+
+    // Compute optimal workspace size
+    cusolverDnDormqr_bufferSize(cusolverH, CUBLAS_SIDE_LEFT, CUBLAS_OP_N, m, n, n, d_A, m, d_tau, d_B, m, &lwork_ormqr);
+
+    // Allocate workspace
+    //work.resize(lwork);
+    //cudaMalloc((void**)&work[0], lwork * sizeof(float));
+
+    // Perform QR factorization and solve the least squares problem
+    //cusolverDnSormqr(cusolverH, CUBLAS_SIDE_LEFT, CUBLAS_OP_N, m, 1, n, d_A, m, nullptr, d_B, m, nullptr, lwork, devInfo);
+}
```

Comment:
Why commented out?

Reply:
The "something" and "something1" tests were just my little checks for making sure that cuda stuff works. Got rid of them